rfqmsg/records.go

```diff
@@ -82,22 +84,43 @@ func (h *Htlc) Balances() []*AssetBalance {
 	return h.Amounts.Val.Balances
 }
 
+// SpecifierChecker is an interface that contains methods for checking certain
+// properties related to asset specifiers.
+type SpecifierChecker interface {
```

Comment:
IIRC we have another instances of this elsewhere in the codebase? 

Also is this something that can just be a function, or do we expect the actual implementation to vary depending on context?